spr/src/commands/init.rs

```diff
@@ -74,7 +74,27 @@ pub async fn init() -> Result<()> {
         return Err(Error::new("Cannot continue without an access token."));
     }
 
+    let github_api_domain = dialoguer::Input::<String>::new()
+        .with_prompt("Github API domain (override for Github Enterprise)")
+        .with_initial_text(
+            config
+                .get_string("spr.githubApiDomain")
+                .ok()
+                .unwrap_or_else(|| "api.github.com".to_string()),
```

Comment:
It seems we're special-casing `api.github.com` consistently, almost using it as a sentinel value, as opposed to actually using the value itself. (i.e., every place we read `spr.githubApiDomain` we check for `api.github.com` and then do something different if that is the value, as opposed to using that value itself.)

So instead, can we make this an `Option<string>`, with `None` meaning "the default non-enterprise", and `Some<x>` meaning "use `x` as the enterprise base URL" -- since we are special-casing `api.github.com` 100% of the time anyway? That seems more consistent with what is actually going on here.